src/Microsoft.ML.Sweeper/AsyncSweeper.cs

```diff
@@ -193,13 +194,13 @@ public sealed class Arguments
         public DeterministicSweeperAsync(IHostEnvironment env, Arguments args)
         {
             _host = env.Register("DeterministicSweeperAsync", args.RandomSeed);
-            _host.CheckUserArg(args.Sweeper.IsGood(), nameof(args.Sweeper), "Please specify a sweeper");
+            _host.CheckValue(args.Sweeper, nameof(args.Sweeper), "Please specify a sweeper");
```

Comment:
>CheckValue [](start = 18, length = 10)

why is this changed to CheckValue from CheckUserArgs?

Reply:
SubComponents have an IsGood() extension method that check both for null and for a non-empty "Kind" property.  CheckUserArg only has an overload for bool, which IsGood() returns.

Now that this is an IComponentFactory, the only thing we can check for is null - which is what CheckValue does.